modules/order/src/OrderTotalSummary.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Drupal\commerce_order;
+
+use Drupal\commerce_order\Entity\OrderInterface;
+
+class OrderTotalSummary implements OrderTotalSummaryInterface {
+
+  /**
+   * The adjustment type manager.
+   *
+   * @var \Drupal\commerce_order\AdjustmentTypeManager
+   */
+  protected $adjustmentTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(AdjustmentTypeManager $adjustment_type_manager) {
+    $this->adjustmentTypeManager = $adjustment_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildTotals(OrderInterface $order) {
+    $total = $order->getTotalPrice();
+    $subtotal = $order->getSubtotalPrice();
+    $collected_adjustments = $order->collectAdjustments();
+    $types = $this->adjustmentTypeManager->getDefinitions();
```

Comment:
I guess it makes sense to move the type check to Adjustment::__construct and remove it from collectAdjustments(). That way we explode right away instead of silently ignoring broken data, probably a better idea.